### PR TITLE
Conditionally avoid method calls deprecated in PHP8

### PR DESCRIPTION
**Pull request by Megatherium**
This small fix is needed to make the actual framework PHP8 compatible as else you'd be flooded in deprecation messages when running the tests.

### CODE REVIEW

**Comment by falkenhawk** [comment]
I wonder why `composer self-update` bumped composer to 2.x (causing jobs to fail) on most of the jobs, except https://travis-ci.com/github/zf1s/phpunit/jobs/423507234 - where it got bumped to 1.10.17 using the same command - and the job succeeded 🤔

**Comment by falkenhawk** [comment]
https://github.com/zf1s/phpunit/pull/3

**Comment by glensc** [comment]
@falkenhawk fix for self-update not updating to v2 was added in: 1.10.10
- https://github.com/composer/composer/releases/tag/1.10.10

and Travis is just ... old!

File: PHPUnit/Util/Class.php [hunk 1]
```diff
@@ -149,24 +149,19 @@ public static function getMethodParameters($method, $forCall = FALSE)
             $typeHint  = '';
 
             if (!$forCall) {
-                if ($parameter->isArray()) {
-                    $typeHint = 'array ';
-                }
-
-                else if (version_compare(PHP_VERSION, '5.4', '>') &&
+                if (PHP_VERSION_ID > 79999 ) {
```
[the remark this answers is not in the document]
Comment (Megatherium) [reply PHPUnit/Util/Class.php:152]: To me they read the same and I don't feel strongly about it either way. 79999 gets a lot of usage in the patch for the framework itself.

Want me to change it everywhere?

[the remark this answers is not in the document]
Comment (glensc) [reply PHPUnit/Util/Class.php:152]: saying "yes" would mean "everywhere" could be interpreted differently.

ok to change the code you added, but leave the rest alone?

disclaimer: this is just my opinion. I'm not a maintainer here.

[the remark this answers is not in the document]
Comment (falkenhawk) [reply PHPUnit/Util/Class.php:152]: I'd also prefer `PHP_VERSION_ID >= 80000` since it easier to read that it affects php 8.0

**Comment by falkenhawk** [comment]
@Megatherium I "fixed" travis builds in #3, now merged. Could you rebase, please?

**Comment by glensc** [comment]
Please fix PR title, the title should describe what does one get when they merge the PR. "small fixes" is as useless as "fixes things".

Useful to read:
- https://chris.beams.io/posts/git-commit/

**Comment by Megatherium** [comment]
Can I be of further assistance or is everything ok now?

**Review by falkenhawk — APPROVED** [review]

**Comment by falkenhawk** [comment]
@Megatherium, thanks for the ping, all good for me, thank you!

**Comment by falkenhawk** [comment]
Released as 3.7.39. zf1s build scripts should pick this version automatically

**Pull request merged**